src/components/PostsList/PostsList.jsx

```diff
@@ -1,37 +1,56 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
 import './PostsList.scss';
 
-export const PostsList = () => (
+export const PostsList = ({
+  posts,
+  selectedPostId,
+  setSelectedPostId,
+  toggleDetail,
+  setToggleDetail,
+}) => (
   <div className="PostsList">
     <h2>Posts:</h2>
 
     <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
-
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
+      {posts.map(post => (
+        <li
+          key={post.id}
+          className="PostsList__item"
         >
-          Open
-        </button>
-      </li>
+          <div>
+            <b>{`[User #${post.userId}]: `}</b>
+            {post.title}
+          </div>
+          <button
+            type="button"
+            className="PostsList__button button"
+            onClick={() => {
+              setSelectedPostId(post.id);
+              if (selectedPostId === post.id) {
+                setToggleDetail(!toggleDetail);
+              } else {
+                setToggleDetail(true);
+              }
+            }}
+          >
+            {(selectedPostId === post.id && toggleDetail) ? 'Close' : 'Open'}
```

Comment:
I believe you can get rid of toggleDetail and setToggleDetail. it does the same function as `selectedPostId === post.id`